core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -195,6 +195,22 @@
   public abstract T idleTimeout(long value, TimeUnit unit);
 
   /**
+   * Sets the maximum message size allowed to be received on the channel. If not called,
+   * defaults to 4 MiB. The default provides protection to clients who haven't considered the
+   * possibility of receiving large messages while trying to be large enough to not be hit in normal
+   * usage.
+   *
+   * <p>This method is advisory, and implementations may decide to not enforce this.  Currently,
+   * the only known transport to not enforce this is {@code InProcessTransport}.
+   *
+   * @param max the maximum number of bytes a single message can be.
+   *
+   * @throws IllegalArgumentException if max is negative.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2307")
+  public abstract T maxInboundMessageSize(int max);
```

Comment:
Hmm, this is trickier than I thought.  To return a `T`, I need to call `thisT` which is on the base class.   Does this casting look right to you?